packages/ckeditor5-source-editing/src/sourceediting.js

```diff
@@ -269,14 +282,17 @@ export default class SourceEditing extends Plugin {
 	}
 
 	/**
-	 * Focuses the textarea containing document source from the first editing root.
+	 * Focuses the textarea containing document source from the first editing root. Places the cursor at the beginning of the textarea.
 	 *
 	 * @private
 	 */
 	_focusSourceEditing() {
 		const [ domSourceEditingElementWrapper ] = this._replacedRoots.values();
 
-		domSourceEditingElementWrapper.querySelector( 'textarea' ).focus();
+		const textarea = domSourceEditingElementWrapper.querySelector( 'textarea' );
+
+		textarea.focus();
+		textarea.selectionEnd = 0;
```

Comment:
Is there a ticket for it? I guess that this PR fixes something else as well.

Reply:
I have not found a ticket, but IMHO it would be cool if switching between modes set the cursor in the same way: at the beginning of the content. I think that such a small improvement does not need to be extracted to a separate issue.